### DISCUSSION TITLE
argocd hard refresh timeout

### DISCUSSION THREAD

**Discussion by imcco** [argoproj/argo-cd#17538] [Q&A]
This week, I updated argocd, 2.5->2.9. I followed the documentation and used CMP to update my plug-in, but there was something that bothered me. When I clicked the force refresh button, it often timed out. I did not encounter this problem in version 2.5. , how should I troubleshoot and optimize. please help me

**Comment by marxus** [comment]
Sorry to inform you it's a problem with the way CMP on a sidecar is designed. 
if you have a large monorepo and you trigger a refresh via a commit it will refresh all the Apps related to that repo, causing Argo to copy the entire repo N amount of times (N being the number of your apps) over to the sidecar, which causes heavy Disk I/O and basically chocking the system.
Unlike the old CMP system which didn't copy the repo N amount of times and allowed concurrent processing.

You can:
1. Scope the refresh using the manifests-path annotation
2. Move out of a monorepo.
